elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexWorkVisitor.java

```diff
@@ -66,6 +68,25 @@
  */
 class ElasticsearchIndexWorkVisitor implements IndexWorkVisitor<Void, BackendRequest<?>> {
 
+	/*
+	 * Nesting marker field name: fields with this name have the parent path as their value,
+	 * and are used for denoting the parent element of the subsequent fields
+	 *
+	 * Use a very specific String to avoid naming conflict with user-defined field names
+	 */
+	public static final String NESTING_MARKER = "__Hibernate_nesting_marker";
```

Comment:
Maybe we should follow the convention used in `ProjectionConstants`? Something like `__HSearch_NestingMarker` and `__HSearch_ThisMarker` - not sure about the camel casing, depends on what you want to convey? I know `ProjectionConstants` is not entirely consistent but it would follow the rule of the majority.